src/helpers/propOr.js

```diff
@@ -0,0 +1,25 @@
+/** @license ISC License (c) copyright 2017 original and current authors */
+/** @author Ian Hofmann-Hicks (evil) */
+
+const curry = require('../core/curry')
+const isNil= require('../core/isNil')
+const isInteger = require('../core/isInteger')
+const isString = require('../core/isString')
+
+const lift = (x, def) =>
+  isNil(x) ? def : x
+
+// propOr : a -> String | Number -> b -> c
+function propOr(def, key, target) {
+  if(!(isString(key) || isInteger(key))) {
+    throw new TypeError('propOr: String or integer required for second argument')
+  }
+
+  if(isNil(target)) {
+    return def
+  }
+
+  return lift(target[key], def)
```

Comment:
I am thinking that we should let null through if the value is set to null. I am okay with guarding for `unit` (`undefined`), but null could be a valid value. What are your thoughts?

Reply:
Yeah, I was thinking about that. I thought to consider `null` as not a valid one to not get possible errors. Maybe this should be a problem on `propPathOr` if one of the father props is null. But on `propOr` I will consider null as a value, make more sense. But on the `pathPropOr` what you think is better? JS Throwing the error or checking if the target is null and return it?

Reply:
I think your implementation is correct. And the other implementation are not.
I Updated prop and propPath to match what you are doing here.